### PR TITLE
add tsan CI

### PR DESCRIPTION
**Pull request by aut0**
Added ThreadSanitizer to CI.
Test failing because of TSAN violation in `403 - systemc/kernel/sc_suspend/Release`

### CODE REVIEW

File: .github/workflows/tsan.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          sudo apt-get update && sudo apt-get install -y qemu-user-static
+        fi
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Build
+      run: |
+        docker buildx build --platform ${{ matrix.platform }} -t systemc_test --build-arg UBUNTU_VERSION=${{ matrix.version }} -f docker/ubuntu.dockerfile .
+        docker run -e SYSTEMC_CI_TARGET=${{ matrix.target }} systemc_test
+  almalinux:
+    runs-on: ubuntu-latest
```
Comment (maehne) [inline .github/workflows/tsan.yml:29]: Copy & paste error: `ubuntu-latest` -> `almalinux-latest`?

File: .github/workflows/tsan.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    - name: Install qemu-user-static
+      run: |
+        if [[ "${{ matrix.platform }}" == "linux/arm64" ]]; then
+          sudo apt-get update && sudo apt-get install -y qemu-user-static
+        fi
+    - name: Checkout
+      uses: actions/checkout@v4
+    - name: Build
+      run: |
+        docker buildx build --platform ${{ matrix.platform }} -t systemc_test --build-arg ALMA_VERSION=${{ matrix.version }} -f docker/alma.dockerfile .
```
Comment (maehne) [inline .github/workflows/tsan.yml:46]: `alma.dockerfile` -> `almalinux.dockerfile`?

**Review by maehne — CHANGES_REQUESTED** [review]
There seems to be at least one copy & paste error for the AlmaLinux OS jobs.

File: .github/workflows/tsan.yml (hunk 1 above)
Comment (aut0) [reply .github/workflows/tsan.yml:29]: No, that's the Action runner. The Almalinux container runs on an Ubuntu host.

File: .github/workflows/tsan.yml (hunk 2 above)
Comment (aut0) [reply .github/workflows/tsan.yml:46]: The file is called `alma.dockerfile`: https://github.com/accellera-official/systemc/blob/main/docker/alma.dockerfile

**Review by maehne — APPROVED** [review]
Thanks for the quick clarifications @aut0! Then, this looks good to me.

**Pull request merged**